google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/SubscriberImpl.java

```diff
@@ -252,4 +278,28 @@ private Status onSeekResponse(Offset seekOffset) {
       return e.getStatus();
     }
   }
+
+  @VisibleForTesting
+  void processBatchFlowRequest() {
+    try (CloseableMonitor.Hold h = monitor.enter()) {
+      if (shutdown) return;
+      connection.modifyConnection(
+          connectedSubscriber ->
+              connectedSubscriber.ifPresent(subscriber -> flushBatchFlowRequest(subscriber)));
```

Comment:
Won't this deadlock? flushBatchFlowRequest also tries to grab the mutex. It would be interesting to exercise this on the text, if possible.

Reply:
java locks are reentrant. But we should exercise this in the test, if it is not.

Reply:
Yep, the [Monitor](https://github.com/google/guava/blob/master/guava/src/com/google/common/util/concurrent/Monitor.java#L35) that we use is reentrant. Existing tests do cover this and they pass.